acquire-driver-common/src/simcams/simulated.camera.c

```diff
@@ -217,20 +218,50 @@ simulated_camera_streamer_thread(struct SimulatedCamera* self)
 {
     clock_init(&self->streamer.throttle);
 
+    size_t bytes_of_data = aligned_bytes_of_image(&self->im.shape);
+    uint8_t* data = (uint8_t*)malloc(bytes_of_data);
+    int64_t frame_id = self->im.frame_id;
+
+    const float exposure_time_ms = self->properties.exposure_time_us * 1e-3f;
+
     while (self->streamer.is_running) {
+        if (self->properties.input_triggers.frame_start.enable) {
+            ECHO(lock_acquire(&self->im.lock));
+            while (!self->software_trigger.triggered) {
+                ECHO(condition_variable_wait(
+                  &self->software_trigger.trigger_ready, &self->im.lock));
+            }
+            self->software_trigger.triggered = 0;
+            ECHO(lock_release(&self->im.lock));
+        }
+
+        clock_tic(&self->streamer.throttle);
+
         struct ImageShape full = { 0 };
         uint32_t origin[2] = { 0, 0 };
 
-        ECHO(lock_acquire(&self->im.lock));
-        ECHO(compute_full_resolution_shape_and_offset(self, &full, origin));
+        // compute the full resolution shape and offset
+        {
+            ECHO(lock_acquire(&self->im.lock));
+            ECHO(compute_full_resolution_shape_and_offset(self, &full, origin));
+            ECHO(lock_release(&self->im.lock));
+
+            size_t nbytes = aligned_bytes_of_image(&full);
+            if (nbytes > bytes_of_data) {
```

Comment:
I think the idea is that the resize already gets done in `simcam_set` but I could see this idea breaking easily since the logic is separated. Might be better to be safe here.

Reply:
I'm just going to eliminate the resize altogether.